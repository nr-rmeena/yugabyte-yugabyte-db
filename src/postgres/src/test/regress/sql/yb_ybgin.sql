--
-- Yugabyte-owned test for ybgin access method.
--

--
-- Create non-temp table which uses Yugabyte storage.
--
CREATE TABLE vectors (i serial PRIMARY KEY, v tsvector);

--
-- tsvector
--
INSERT INTO vectors (v) VALUES
    (to_tsvector('simple', 'a bb ccc')),
    (to_tsvector('simple', 'bb a e i o u')),
    (to_tsvector('simple', 'ccd'));
CREATE INDEX ON vectors USING ybgin (v);

-- Cleanup
DROP TABLE vectors;

--
-- Try creating ybgin index on temp table.
--
CREATE TEMP TABLE vectors (i serial PRIMARY KEY, v tsvector);
CREATE INDEX ON vectors USING ybgin (v);

-- Cleanup
DISCARD TEMP;
