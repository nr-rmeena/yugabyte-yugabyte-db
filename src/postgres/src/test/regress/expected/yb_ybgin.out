--
-- Yugabyte-owned test for ybgin access method.
--
--
-- Create non-temp table which uses Yugabyte storage.
--
CREATE TABLE vectors (i serial PRIMARY KEY, v tsvector);
--
-- tsvector
--
INSERT INTO vectors (v) VALUES
    (to_tsvector('simple', 'a bb ccc')),
    (to_tsvector('simple', 'bb a e i o u')),
    (to_tsvector('simple', 'ccd'));
CREATE INDEX ON vectors USING ybgin (v);
ERROR:  index method "ybgin" not supported yet
HINT:  See https://github.com/YugaByte/yugabyte-db/issues/1337. Click '+' on the description to raise its priority
-- Cleanup
DROP TABLE vectors;
--
-- Try creating ybgin index on temp table.
--
CREATE TEMP TABLE vectors (i serial PRIMARY KEY, v tsvector);
CREATE INDEX ON vectors USING ybgin (v);
ERROR:  access method "ybgin" only supported for indexes using Yugabyte storage
-- Cleanup
DISCARD TEMP;
